pkg/ottl/ottlfuncs/func_is_in_cidr.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type IsInCIDRArguments[K any] struct {
+	Target    ottl.StringGetter[K]
+	Addresses []string
```

Comment:
```suggestion
	Networks []ottl.StringGetter[K]
```

Let's make it more dynamic by not restricting it to string literals.